### PR TITLE
Android lint to NNBD

### PR DESCRIPTION
**Pull request by dnfield**
See https://github.com/flutter/engine/pull/23151 for context

Fixes https://github.com/flutter/flutter/issues/72542

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: refactor with no semantic change

**Review by chinmaygarde — APPROVED** [review]

**Review by hamad22h — COMMENTED** [review]
[]()

File: tools/android_lint/bin/main.dart [hunk 1]
```diff
@@ -2,11 +2,227 @@
 // Use of this source code is governed by a BSD-style license that can be
```
Comment (hamad22h) [inline tools/android_lint/bin/main.dart:2]: انا

**Review by hamad22h — COMMENTED** [review]
[]()

**Pull request merged**